src/library/scala/util/Properties.scala

```diff
@@ -71,44 +71,24 @@ private[scala] trait PropertiesTrait {
   def scalaPropOrEmpty(name: String): String             = scalaPropOrElse(name, "")
   def scalaPropOrNone(name: String): Option[String]      = Option(scalaProps.getProperty(name)).orElse(propOrNone("scala." + name))
 
-  /** The numeric portion of the runtime Scala version, if this is a final
-   *  release.  If for instance the versionString says "version 2.9.0.final",
-   *  this would return Some("2.9.0").
-   *
-   *  @return Some(version) if this is a final release build, None if
-   *  it is an RC, Beta, etc. or was built from source, or if the version
-   *  cannot be read.
+  /** The version the Scala runtime, if this is not a snapshot.
    */
-  val releaseVersion =
-    for {
-      v <- scalaPropOrNone("maven.version.number")
-      if !(v endsWith "-SNAPSHOT")
-    } yield v
-
-  /** The development Scala version, if this is not a final release.
-   *  The precise contents are not guaranteed, but it aims to provide a
-   *  unique repository identifier (currently the svn revision) in the
-   *  fourth dotted segment if the running version was built from source.
-   *
-   *  @return Some(version) if this is a non-final version, None if this
-   *  is a final release or the version cannot be read.
+  val releaseVersion = scalaPropOrNone("maven.version.number").filterNot(_.endsWith("-SNAPSHOT"))
+
+  /** The version the Scala runtime, if this is a snapshot.
    */
-  val developmentVersion =
-    for {
-      v <- scalaPropOrNone("maven.version.number")
-      if v endsWith "-SNAPSHOT"
-      ov <- scalaPropOrNone("version.number")
-    } yield ov
-
-  /** Either the development or release version if known, otherwise
-   *  the empty string.
+  val developmentVersion = scalaPropOrNone("maven.version.number").filter(_.endsWith("-SNAPSHOT")).flatMap(_ => scalaPropOrNone("version.number"))
+
+  /** The version the Scala runtime, or the empty string if unknown.
+   *
+   *  Note that the version of the Scala library need not correlate with the version of the Scala compiler
+   *  used to emit either the library or user code.
```

Comment:
wdyt about also adding something like:

> For example, Scala 3.0 and 3.1 (and perhaps future 3.x versions) use the Scala 2.13 standard library as-is, so `versionString` will return a string such as "2.13.6" rather than the Scala 3 version you might have expected

Also, is there anything we can say about how Scala 3 users can find out what version of scala3-library they're running?

Reply:
That crossed my mind, but it seemed heavy-handed to over-specify in library scaladoc. But perhaps that is too fussy. I don't know on scala 3, but I'll create a ticket for API to return library and compiler versions.

Reply:
I made it sweeter by keeping it shorter.

Reply:
oh actually they copied over Properties to Dotty, I guess because of the old dotty t-shirt, "If it's in Scala 2, it must not be good enough." I added that hint.